examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,233 @@
+import { DataSource, syncCollection } from "./data"
+
+import { ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { useState, useMemo, useEffect, memo } from "react"
+import { computeFieldsFromDataSource, mergeFieldsWithExistingFields } from "./data"
+import { Spinner } from "./components/Spinner"
+
+function ChevronIcon() {
+    return (
+        <svg xmlns="http://www.w3.org/2000/svg" width="8" height="16">
+            <path
+                d="M 3 11 L 6 8 L 3 5"
+                fill="transparent"
+                strokeWidth="1.5"
+                stroke="#999"
+                strokeLinecap="round"
+                strokeLinejoin="round"
+            />
+        </svg>
+    )
+}
+
+interface FieldMappingRowProps {
+    field: ManagedCollectionField
+    isIgnored: boolean
+    onFieldToggle: (fieldId: string) => void
+    onFieldNameChange: (fieldId: string, name: string) => void
+}
+
+const FieldMappingRow = memo(({ field, isIgnored, onFieldToggle, onFieldNameChange }: FieldMappingRowProps) => {
+    const [hasCustomName, setHasCustomName] = useState(field.id !== field.name)
+
+    return (
+        <>
+            <div
+                className="source-field"
+                aria-disabled={isIgnored}
+                onClick={() => onFieldToggle(field.id)}
+                role="button"
+            >
+                <input
+                    type="checkbox"
+                    checked={!isIgnored}
+                    onChange={event => {
+                        event.stopPropagation()
+                    }}
+                />
+                <span>{field.id}</span>
+            </div>
+            <ChevronIcon />
+            <input
+                type="text"
+                style={{
+                    width: "100%",
+                    opacity: isIgnored ? 0.5 : 1,
+                }}
+                disabled={isIgnored}
+                placeholder={field.id}
+                value={hasCustomName ? field.name : ""}
+                onChange={event => {
+                    const value = event.target.value
+                    if (!value.trim()) {
+                        setHasCustomName(false)
+                        onFieldNameChange(field.id, field.id)
+                    } else {
+                        setHasCustomName(true)
+                        onFieldNameChange(field.id, value.trimStart())
+                    }
+                }}
+            />
+        </>
+    )
+})
+
+interface FieldMappingProps {
+    collection: ManagedCollection
+    dataSource: DataSource
+    initialSlugFieldId: string | null
+}
+
+type FieldMappingStatus = "mapping-fields" | "loading-fields" | "syncing-collection"
+
+export function FieldMapping({ collection, dataSource, initialSlugFieldId }: FieldMappingProps) {
```

Comment:
Nit. Since the Props are so important to the component I would move that next to the component and have other types like `FieldMappingStatus` be further away. I think it would also be fine to have that type inline since it isn't used anywhere else